osu.Game/Online/Leaderboards/LeaderboardManager.cs

```diff
@@ -183,11 +186,31 @@ private void localScoresChanged(IRealmCollection<ScoreInfo> sender, ChangeSet? c
                 }
             }
 
-            newScores = newScores.Detach().OrderByCriteria(CurrentCriteria.Sorting);
+            newScores = newScores.Detach();
+
+            if (CurrentCriteria.Sorting == LeaderboardSortMode.PP) updatePpValues(newScores);
+
+            newScores = newScores.OrderByCriteria(CurrentCriteria.Sorting);
 
             var newScoresArray = newScores.ToArray();
             scores.Value = LeaderboardScores.Success(newScoresArray, newScoresArray.Length, null);
         }
+
+        private void updatePpValues(IEnumerable<ScoreInfo> scores)
```

Comment:
why

Reply:
explain it. inline. or in pr. or anything really.

Reply:
I've added the comment that says "score.PP value is never actually set and saved for scores present in database. So just as with other cases where it's used (such as BeatmapLeaderboardScore_Tooltip and ScorePanel PerformanceStatistics) - it has to be calculated in real time before being used".

I've checked many scores personally - and they never contain the .PP value. All those categories of scores have .PP == null:
- Unranked scores set on a client
- Ranked scores set on a client
- Ranked scores imported from server
- Any scores imported from .osr file

Reply:
I've added the comment that says "score.PP value is never actually set and saved for scores present in database. So just as with other cases where it's used (such as BeatmapLeaderboardScore_Tooltip and ScorePanel PerformanceStatistics) - it has to be calculated in real time before being used".

I've checked many scores personally - and they never contain the .PP value. All those categories of scores have .PP == null (at least on the ScoreInfo that's passed to the song select v2):
- Unranked scores set on a client
- Ranked scores set on a client
- Ranked scores imported from server
- Any scores imported from .osr file